narwhal/worker/src/handlers.rs

```diff
@@ -277,19 +293,39 @@ impl<V: TransactionValidator> PrimaryToWorker for PrimaryReceiverHandler<V> {
             while let Some(result) = handles.next().await {
                 match result {
                     Ok(response) => {
-                        if let Some(batch) = response.into_body().batch {
+                        if let Some(mut batch) = response.into_body().batch {
                             if !message.is_certified {
                                 // This batch is not part of a certificate, so we need to validate it.
-                                if let Err(err) = self.validator.validate_batch(&batch).await {
+                                if let Err(err) = self
+                                    .validator
+                                    .validate_batch(&batch, &self.protocol_config)
+                                    .await
+                                {
                                     // The batch is invalid, we don't want to process it.
                                     return Err(anemo::rpc::Status::new_with_message(
                                         StatusCode::BadRequest,
                                         format!("Invalid batch: {err}"),
                                     ));
                                 }
                             }
+
+                            // TODO: Remove once we have upgraded to protocol version 12.
+                            validate_batch_version(&batch, &self.protocol_config).map_err(
```

Comment:
Is this validation necessary? I thought it is already done inside `validate_batch()` above.

Reply:
Unfortunately we only run validate_batch on header batches or this would have been a much cleaner change but I had to add it here for the certificates batch